src/lib/translator.ts

```diff
@@ -0,0 +1,117 @@
+import { CalendarEventObject, Attendee, Dictionary, RRule } from '../types/dav-parser';
+import { Component, Property } from 'ical.js';
+
+/**
+ * Translates a CalendarEventObject into an ICS String.
+ * 
+ * @param event CalendarEventObject - the event object to convert to ICS
+ * @returns String  -  the ICS string representing the event
+ */
+export function translate(event: CalendarEventObject): string {
+  const calendarComponent: Component = new Component('vcalendar');
+  const eventComponent: Component = buildEventComponent(event);
+
+  calendarComponent.addPropertyWithValue('version', '2.0');
+  calendarComponent.addPropertyWithValue('calscale', 'GREGORIAN');
+  // add the event component as a subcomponent for the calendar component
+  calendarComponent.addSubcomponent(eventComponent);
+
+  // transform to ICS
+  return calendarComponent.toString();
+}
+
+/**
+ * transforms an event object into a ICAL component
+ * 
+ * @param event CalendarEventObject - the event object
+ * @returns Component the ICAL component representing the event
+ */
+const buildEventComponent = (event: CalendarEventObject): Component => {
+  const eventComponent = new Component('vevent');
+
+  eventComponent.addPropertyWithValue('uid', event.id);
+  eventComponent.addPropertyWithValue('summary', event.title);
+  eventComponent.addPropertyWithValue('location', event.location || '');
+  eventComponent.addPropertyWithValue('description', event.description || '');
+  eventComponent.addPropertyWithValue('dtstart', event.start as string);
+  eventComponent.addPropertyWithValue('dtend', event.end as string);
```

Comment:
Why are you assuming `event.start` and `event.date` will always be `string`s? They can also be `Date`s, can't they?